native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -87,16 +88,16 @@ public static Object toJson(BXml xml, BMap<?, ?> options) {
      * @param preserveNamespaces preserve the namespaces when converting
      * @return JSON representation of the given xml object
      */
-    public static Object convertToJSON(BXml xml, String attributePrefix, boolean preserveNamespaces) {
-
+    public static Object convertToJSON(BXml xml, String attributePrefix, boolean preserveNamespaces,
+                                       AttributeManager attributeManager) {
         if (xml instanceof BXmlItem) {
-            return convertElement((BXmlItem) xml, attributePrefix, preserveNamespaces);
+            return convertElement((BXmlItem) xml, attributePrefix, preserveNamespaces, attributeManager);
         } else if (xml instanceof BXmlSequence) {
             BXmlSequence xmlSequence = (BXmlSequence) xml;
             if (xmlSequence.isEmpty()) {
-                return newJsonList();
+                return StringUtils.fromString(EMPTY_STRING);
```

Comment:
Is there any possibility for this scenario? if the XML sequence is empty, won't it be considered an element?

Reply:
I couldn't figure-out the use case for this but this is an iteration method for complex xml. So I keep this as we have a chance to get empty sequence by coding wise.

Reply:
Ack